Documentation (#4)
cybozu-go/pod-security-admission: docs/configuration.md

### UserProfile

| Name              | Type | Description                                          |
| ----------------- | ---- | ---------------------------------------------------- |
| denyRunAsRoot     | bool | Deny running as root users                           |
| forceRunAsNonRoot | bool | Force running with non-root users by MutatingWebhook |

Customize Profile
-----------------

By default, `pod-security-admission` uses the following configuration to enforce [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/):

```yaml
- name: baseline
  denyHostNamespace: true
  denyPrivilegedContainers: true
  capabilities:
    denyUnsafeCapabilities: true
  volumes:
    denyHostPathVolumes: true
  hostPorts:
    denyHostPorts: true
  denyUnsafeApparmor: true
  denyUnsafeSelinux: true
  denyUnsafeProcMount: true
  denyUnsafeSysctls: true
- name: restricted
  volumes:
    denyNonCoreVolumeTypes: true
  denyPrivilegeEscalation: true
  users:
    denyRunAsRoot: true
  denyRootGroups: true
  denyUnsafeSeccomp: true
```

For example, administrators can customize the profile under the `Baseline` section as follows.

> This profile allows to add `SYSLOG` and `NET_ADMIN` capability, use of hostPort from 1024 to 65535.

Reviewer: ```suggestion
This profile allows users to add `SYSLOG` and `NET_ADMIN` capabilities, use of hostPort from 1024 to 65535.
```

| forceRunAsNonRoot        | bool        | Force running with non-root users by MutatingWebhook                                                                                        |
| rootGroups               | bool        | Allow running with a root primary or supplementary GID                                                                                      |
| seccomp                  | bool        | Allow usage of non-default Seccomp profile                                                                                                  |

### PortRange

| Name | Type  | Description                            |
| ---- | ----- | -------------------------------------- |
| min  | int32 | The min of host port range (inclusive) |
| max  | int32 | The max of host port range (inclusive) |

Customize Profile
-----------------

By default, `pod-security-admission` uses the following configuration to enforce [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/):

> The profile is described only the items to be allowed. A false or no description indicates that the item is to be denied.

Reviewer 2: ```suggestion
The profile describes only the allowed items. A false or no description indicates that the item is to be denied.
```

```yaml
- name: baseline
  nonCoreVolumeTypes: true
  allowPrivilegeEscalation: true
  runAsRoot: true
  rootGroups: true
  seccomp: true
- name: restricted
  forceRunAsNonRoot: true
```

> For example, administrators can customize the profile.

Reviewer 2: ```suggestion
Administrators can customize the profile if needed.
```
